src/App.tsx

```diff
@@ -1,27 +1,53 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
-
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+// import { all } from 'cypress/types/bluebird';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  type Callback = () => Promise<Good[]>;
+
+  const setSelectedGoodsCallback = (getGoods: Callback) => {
+    getGoods()
+      .then((products: Good[]) => setGoods(products));
+  };
+
+  const setSelectedGoods = useCallback(setSelectedGoodsCallback, [goods]);
```

Comment:
why do you need it?
Just use useCallback in your `setSelectedGoodsCallback` function